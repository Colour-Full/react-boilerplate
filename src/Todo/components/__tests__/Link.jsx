// @flow

import React from 'react'
import type Link from '../Link'
import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

Enzyme.configure({ adapter: new Adapter() })

describe('A Link Component ', () => {
  it('It should render without throwing an error ', () => {
    const wrapper = shallow(<Link />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should call onClick handler when button is clicked', () => {
    const spy = jest.fn()
    const wrapper = shallow(<Link onClick={spy} />)
    wrapper.find('button').simulate('click')
    expect(spy).toHaveBeenCalledTimes(1)
  })
})
