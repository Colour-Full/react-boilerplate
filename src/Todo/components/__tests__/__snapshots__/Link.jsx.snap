// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Link Component  It should render without throwing an error  1`] = `
<button
  style={
    Object {
      "marginLeft": "4px",
    }
  }
/>
`;
